compliance/collection/nodeinventorizer/nodeinventory.go

```diff
@@ -65,6 +77,50 @@ func (n *NodeInventoryCollector) Scan(nodeName string) (*storage.NodeInventory,
 	return m, nil
 }
 
+// callAnalyzer isolates the call to Analyze to ensure we recover from any panic that might occur.
+// Analyze runs on a potentially unknown live file system, so it might be prone to unexpected behavior.
+func (n *NodeInventoryCollector) callAnalyzer(nodeName string) (*nodes.Components, error) {
+
+	type analyzerResult struct {
+		components *nodes.Components
+		err        error
+	}
+
+	results := make(chan analyzerResult)
+
+	go func() {
+		defer func(c chan analyzerResult) {
+			r := recover()
+			if r != nil {
+				log.Warnf("Analyzer call panicked: %v", r)
+				c <- analyzerResult{
+					components: nil,
+					err:        fmt.Errorf("call to Analyzer function panicked: %v", r),
+				}
+			}
+			close(c)
+		}(results)
+
+		// uncertifiedRHEL is set to false, as scans are only supported on RHCOS for now,
+		// which only exists in certified versions
+		c, err := n.analyzerFunction(nodeName, "/host/", nodes.AnalyzeOpts{UncertifiedRHEL: false, IsRHCOSRequired: true})
+		results <- analyzerResult{
+			components: c,
+			err:        err,
+		}
```

Comment:
I see, we have forgotten about one aspect. What we do now, we call the `analyzerFunction` and wait for the result for `analyzerCallDeadline` time. When we miss the deadline, we simply ignore the function's result and return an error. However, the function will still be running and can potentially consume the node resources. 

I think we should pass the context into the `analyzerFunction` (this includes also passing it to `nodes.Analyze`) and make it stop when the context is canceled. This would allow us to protect both:
- the ACS component running this (be it compliance or nodeScanner)
- the customer's node

Reply:
@Maddosaurus Maybe worth looking into https://github.com/stackrox/stackrox/blob/master/pkg/safe/run.go#L38

Reply:
To address this issue, we first need to add context support to `nodes.Analyze` - here is a PR for this: https://github.com/stackrox/scanner/pull/1118